### DISCUSSION TITLE
Solidifying classes

### DISCUSSION THREAD

**Discussion by AlixANNERAUD** [berry-lang/berry#320] [Q&A]
I have a question about solidifying C++ classes into Berry classes. I haven't found a way to solidify C++ classes "directly" into Berry. Am I missing something, or is it not possible to do it with solidify? Thanks in advance for your answer.

**Comment by s-hadinger** [comment]
I'm not sure what you are asking for. It's the opposite. You soldifify Berry into C to embed in Flash.

If you want to easily access C code from Berry, take a look at https://github.com/berry-lang/berry_mapping

**Comment by AlixANNERAUD** [comment]
Forgive me for not being clear enough. What I would like is to solidify classes (C++) in Berry. That is to say that from a class, soldify create me the corresponding berry class (with the corresponding attributes) and make me the binding of the methods. It is true that this is not a trivial operation and that from what I have seen, in Berry, it is only possible to do function bidding or in the case of classes, to use functions in which we pass a pointer to the object as an argument. However, this is a laborious operation to do. So, I would like to know if there is any other method?

**Reply by s-hadinger** [reply]
That's not possible per se. C++ has a level of features and complexity far beyond Berry. You can't just map a C++ class to a Berry class. 

Take a look at Berry mapping lib, it provides the primitives to have a lightweight mapping to C++. But nothing automatic
